fv3core/fv3core/initialization/dycore_state.py

```diff
@@ -0,0 +1,380 @@
+                raise KeyError(
+                    variable_name + " is provided, but not part of the dycore state"
+                )
+            getattr(state, variable_name).data[:] = data
+        for field_name in field_names:
+            if field_name not in dict_of_numpy_arrays.keys():
+                raise KeyError(
+                    field_name
+                    + " is not included in the provided dictionary of numpy arrays"
+                )
+        return state
+
+    @classmethod
+    def init_from_quantities(cls, dict_of_quantities):
+        field_names = [field.name for field in fields(cls)]
+        for variable_name, data in dict_of_quantities.items():
+            if variable_name not in field_names:
+                raise KeyError(
+                    variable_name + " is provided, but not part of the dycore state"
+                )
+        for field_name in field_names:
+            if field_name not in dict_of_quantities.keys():
+                raise KeyError(
+                    field_name
+                    + " is not included in the provided dictionary of quantities"
+                )
+            elif not isinstance(dict_of_quantities[field_name], fv3util.Quantity):
+                raise TypeError(
+                    field_name
+                    + " is not a Quantity, but instead a "
+                    + str(type(dict_of_quantities[field_name]))
+                )
+        for _field in fields(cls):
+            for varcheck in ["units", "dims"]:
+                if (
+                    _field.metadata[varcheck]
+                    != dict_of_quantities[_field.name][varcheck]
+                ):
+                    raise TypeError(
+                        field_name
+                        + " has metadata "
+                        + varcheck
+                        + " of "
+                        + dict_of_quantities[_field.name][varcheck]
+                        + " that does not match the requirement"
+                        + _field.metadata[varcheck]
+                    )
+        return cls(**dict_of_quantities, quantity_factory=None)
+
+    @classmethod
+    def init_from_serialized_data(cls, serializer, grid, quantity_factory):
+        savepoint_in = serializer.get_savepoint("FVDynamics-In")[0]
+        translate_object = fv3core.testing.TranslateFVDynamics([grid])
+        input_data = translate_object.collect_input_data(serializer, savepoint_in)
+        # making just storages for the moment, revisit when making them all
+        # quantities (maybe use state_from_inputs)
+        translate_object._base.make_storage_data_input_vars(input_data)
+        # used for the translate test as inputs, but are generated by the
+        # MetricsTerms class and are not part of this data class
+        for delvar in ["ak", "bk", "ptop", "ks"]:
+            del input_data[delvar]
+        return cls(**input_data, quantity_factory=quantity_factory)
```

Comment:
This method should only be getting used by the tests, and currently fv3core doesn't depend on serialbox other than for testing. Can this get moved to the testing code?

Reply:
It can. I was thinking it might be useful for performance testing if we end up wanting to test non-baroclinic namelists before we implement reading initial state from disk. But someone wanting to do that can probably figure it out.